packages/framework/src/Model/Product/BestsellingProduct/CachedBestsellingProductFacade.php

```diff
@@ -31,22 +31,25 @@ public function __construct(
      * @param int $domainId
      * @param \Shopsys\FrameworkBundle\Model\Category\Category $category
      * @param \Shopsys\FrameworkBundle\Model\Pricing\Group\PricingGroup $pricingGroup
+     * @param int $limit
      * @return int[]
      */
-    public function getAllOfferedBestsellingProductIds(
+    public function getOfferedBestsellingProductIds(
         int $domainId,
         Category $category,
         PricingGroup $pricingGroup,
+        int $limit,
     ): array {
         $cacheId = $this->getCacheId($domainId, $category, $pricingGroup);
```

Comment:
I recommend adding a limit to the cache ID, so when you call it twice with a different limit, the results will be as expected.